### DISCUSSION TITLE
Long queue

### DISCUSSION THREAD

**Discussion by c0pt4n** [ROM-builders/temporary#16842] [Q&A]
We should take a step against some buildbots that are ruining the server and long queue , not all write access users deserve it cause they spam builds about build everyday.

now average waiting time is 9 hrs.

for ex: 
![Screenshot_20230110-224309_Chrome](https://user-images.githubusercontent.com/87261955/211658851-e069d287-b943-4c62-9f9c-6a399d8c2ad3.png)
imagine 10 hrs waiting and I am still the 13th

**Comment by Apon77** [comment]
What steps can be taken?

**Reply by Aflaungos** [reply]
Bro @Apon77 First off thanks a lot for the free server!

Can you take people's permission to cancel builds away? I believe some people are using it for bad intentions. Maybe just give permission for people you trust, not everyone from team

As for the build bots issue, is not not possible to add a PR timeout, so each pull has to be in between 1h to work? It would reduce spam, but not solve it, maybe if it was possible to add some sort of captcha to confirm PR in Github or outside

**Reply by c0pt4n** [reply]
> What steps can be taken?

it is your choice sir , but some suggestions:
1- clean of the people who has write access to server , make them apply with their trees and you review them as I think it will be obvious if he is noob , buildbot(as some people take another dev tree and only commit the bringup commit and boom! he calls it his tree and build 69 rom on cirrus with it , so it produces 15 hr queue for average for another devs which delays releasing roms)

2- some people ruin the server with unofficial builds also the official build is being built on cirrus!

3- make pr limit 

and thanks for your response sir , I appreciate your work to this community , I hope you can take more aggressive steps against toxic and buildbots

**Reply by mrx7014** [reply]
> What steps can be taken?

Thanks Bro for this free server,and i hope you find a solution to stop these bots from building.

**Reply by SeifHossam17** [reply]
> What steps can be taken?

I know this server free
but why we still here waiting 15 hours
and finally we have errors of this rom and waiting again

**Reply by mrx7014** [reply]
> > What steps can be taken?
> 
> I know this server free but why we still here waiting 15 hours and finally we have errors of this rom and waiting again

Yes bro, I agree with you

**Reply by NRanjan-17** [reply]
> Bro @Apon77 First off thanks a lot for the free server!
> 
> Can you take people's permission to cancel builds away? I believe some people are using it for bad intentions. Maybe just give permission for people you trust, not everyone from team
> 
> As for the build bots issue, is not not possible to add a PR timeout, so each pull has to be in between 1h to work? It would reduce spam, but not solve it, maybe if it was possible to add some sort of captcha to confirm PR in Github or outside

Builds are only cancelled if you are caught doing multiple pr or making pr with multiple accounts.
cancelling builds are only allowed to push users

**Reply by NRanjan-17** [reply]
> What steps can be taken?

1: sir we can remove old and discontinued roms
2: we can remove old version of roms which are not maintained.
3: mechanism to limit pr spam

**Comment by Apon77** [comment]
I actually have no idea what to do or how to do, I almost tried many possible way to use best use of the server. 
May be only way is increasing the resources, but I dont know how to do so.

**Reply by c0pt4n** [reply]
I think you also should make a form and make all push users reapply with their trees , vendor , kernel and you review them and the suitable guy(who seems good in git , no kanger, not buildbotter) will continue but if not good then kick from push users.

**Reply by Apon77** [reply]
that is too much work, i already include in push group by checking git and commit history. 
for buildbot, they will die automatically or will be pro by time.

**Reply by c0pt4n** [reply]
if you need help I can help you because for some maintainers who is bringing fixes and other stuffs it is too hard to wait for 15 hr to test if the build error is gone or no.

I have some examples of builders take for example my tree and only brought Bringing commit and they are push users.
